src/Mod/Fem/Gui/ViewProviderFemPostFilter.cpp

```diff
@@ -49,7 +49,8 @@ ViewProviderFemPostDataAlongLine::~ViewProviderFemPostDataAlongLine() = default;
 void ViewProviderFemPostDataAlongLine::setupTaskDialog(TaskDlgPost* dlg)
 {
     // add the function box
-    dlg->appendBox(new TaskPostDataAlongLine(dlg->getView()));
+    auto view = static_cast<ViewProviderFemPostDataAlongLine*>(dlg->getView());
```

Comment:
isn't `view` equal to `this`? If yes, then it's probably best to assert this and pass `this` to TaskPostDataAlongLine.
At this point it's not known which view provider is passed to TaskDlgPost and thus the static_cast is potentially harmful. It is only expected that a ViewProviderFemPostDataAlongLine was passed.

Reply:
Yes, `view` is equal `this`.
I have reverted the constructor to receive a `Gui::ViewProviderDocumentObject` and to include the original assertion again.
So your suggestion is to move the assert in the TaskPost constructors to the caller, check if `this` is equal to `dlg->getView()` and change the constructors to use the specific pointer instead of `ViewProviderDocumentObject`?
After all, currently only the type is checked but not the pointer to the view provider.

Reply:
> So your suggestion is to move the assert

No, my idea was to do this in ViewProviderFemPostDataAlongLine::setupTaskDialog
```
assert(this == dlg->getView());
dlg->appendBox(new TaskPostDataAlongLine(this));
```